lib/ansible/modules/cloud/misc/terraform.py

```diff
@@ -389,7 +390,11 @@ def main():
     if state == 'absent' and workspace != 'default' and purge_workspace is True:
         remove_workspace(command[0], project_path, workspace)
 
-    module.exit_json(changed=changed, state=state, workspace=workspace, outputs=outputs, stdout=out, stderr=err, command=' '.join(command))
+    if module.check_mode:
+        changed = needs_application
```

Comment:
can you explain, why these 2 lines?

Reply:
Looks like ansible won't display the diff output unless the module report changes so if the plan is not clean (0 resources added, deleted or changed) we set `changed` to `true` (actually directly to `needs_application` that is set by `build_plan` based on the return code of terraform plan but with the same effect). This is useful when we use `--diff` in conjunction with `--check`